cloud_adapters/azure_cloud_connector_adapter/src/azure_cloud_connector_adapter.rs

```
@@ -81,16 +82,22 @@ impl CloudAdapter for AzureCloudConnectorAdapter {
     /// Creates a new instance of a CloudAdapter with default settings
     fn create_new() -> Result<Box<dyn CloudAdapter + Send + Sync>, CloudAdapterError> {
         let cloud_connector_client = futures::executor::block_on(async {
-            let out_dir_path = env!("OUT_DIR");
             let config_file =
-                fs::read_to_string(Path::new(out_dir_path).join(CONFIG_FILE_NAME)).unwrap();
+                fs::read_to_string(Path::new(env!("OUT_DIR")).join(CONFIG_FILE)).unwrap();
 
             // Load the config
-            let config: ConfigItem = serde_json::from_str(&config_file).unwrap();
-
-            AzureCloudConnectorClient::connect(config.cloud_connector_url)
-                .await
-                .map_err(CloudAdapterError::communication)
+            let config: Config = serde_json::from_str(&config_file).unwrap();
```

What happens if a JSON parser occurs here?

---

assuming you mean parser error, this will panic. This needs to use `?` and `map_err` instead

---

If an error occurs here when trying to parse the JSON config, the program should crash immediately.

---

no, we should return the error to the caller, this method returns `Result`. Panics during runtime are bad, we should avoid them unless absolutely necessary

---

In this case, shouldn't we panic here if the configuration file can't be parsed to avoid an unstable state in the program unless we can handle that error gracefully?

---

no, we should return an error. most likely the callers will just bubble this all the way up to main and the program will terminate that way, but we should not panic. if nothing else, this makes this method quite a bit harder to test

---

Removed unwrap() and now the error is returned to the caller.